### PR TITLE
services.ec2: Delete the source for every image urls

### PR DESCRIPTION
**Pull request by sayanchowdhury**
Signed-off-by: Sayan Chowdhury <sayan.chowdhury2012@gmail.com>

### CODE REVIEW

File: fedimg/services/ec2/ec2imguploader.py [hunk 1]
```diff
@@ -24,6 +24,8 @@
 
 import re
 
+from time import sleep
+
```
Comment (dustymabe) [inline fedimg/services/ec2/ec2imguploader.py]: i don't understand why this is removed. things in this file use it

File: fedimg/services/ec2/ec2initiate.py [hunk 2]
```diff
@@ -163,5 +163,8 @@ def main(image_urls, access_id, secret_key, regions, volume_types=None,
             #TODO: Implement the clean up of the images if failed.
             # uploader.clean_up(image_id=image.id, delete_snapshot=True)
 
-    shutil.rmtree(os.path.dirname(source))
+        if source:
```
Comment (dustymabe) [inline fedimg/services/ec2/ec2initiate.py]: in what cases does source evaluate to non true? I don't see what problem this is solving.

Comment (sayanchowdhury) [reply fedimg/services/ec2/ec2initiate.py]: There could be various reasons when source evaluates to non true, euca2ools does not setup, or wget fails.
fedimg needs to be better at raising and catching exceptions. I have opened up #113 for the same.

Comment (dustymabe) [reply fedimg/services/ec2/ec2initiate.py]: it looks like this should be in a `finally` block then. If the valueError gets raised above then this code wouldn't get hit right?

File: fedimg/services/ec2/ec2imguploader.py (hunk 1 above)
Comment (sayanchowdhury) [reply fedimg/services/ec2/ec2imguploader.py]: This was not removed, this was added.

Comment (dustymabe) [reply fedimg/services/ec2/ec2imguploader.py]: oops. i guess my diff view confused me on that. Can you put this in a separate commit?

File: fedimg/services/ec2/ec2initiate.py [hunk 3]
```diff
@@ -158,10 +158,16 @@ def main(image_urls, access_id, secret_key, regions, volume_types=None,
                 published_images.extend(publisher.publish_images(
                     region_image_mapping=[(region, image.id)]
                 ))
+        except ValueError:
+            _log.debug('Source not found.')
+            return []
         except Exception as e:
             _log.debug(e.message)
+            return []
```
Comment (dustymabe) [inline fedimg/services/ec2/ec2initiate.py]: so here we are returning `[]` where we weren't sending anything back before. wehere is this main function getting called from?

File: fedimg/services/ec2/ec2initiate.py [hunk 4]
```diff
@@ -158,10 +158,16 @@ def main(image_urls, access_id, secret_key, regions, volume_types=None,
                 published_images.extend(publisher.publish_images(
                     region_image_mapping=[(region, image.id)]
                 ))
+        except ValueError:
+            _log.debug('Source not found.')
+            return []
```
Comment (dustymabe) [inline fedimg/services/ec2/ec2initiate.py]: why not just do this on line 94 above where we raise the `ValueError` ?

File: fedimg/services/ec2/ec2initiate.py [hunk 5]
```diff
@@ -91,7 +91,8 @@ def main(image_urls, access_id, secret_key, regions, volume_types=None,
         try:
             source = get_source_from_image(image_url)
             if not source:
-                raise ValueError
+                _log.debug('Source not found.')
+                return []
```
Comment (dustymabe) [inline fedimg/services/ec2/ec2initiate.py]: this looks good for now but in the future I hope to get more output in the logs other than `Source not found.`. I opened https://github.com/fedora-infra/fedimg/issues/118 for this

**Comment by dustymabe** [comment]
LGTM

File: fedimg/services/ec2/ec2initiate.py (hunk 5 above)
Comment (sayanchowdhury) [reply fedimg/services/ec2/ec2initiate.py]: Thanks!

**Pull request merged**